plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/dialog/SelectRouteDialog.java

```diff
@@ -28,7 +32,25 @@
 	public SelectRouteDialog(List<IRoute> routes, Shell shell) {
 		super(shell, new RouteLabelProvider());
 		setTitle("Select Route");
-		setMessage("Select the route to open in a browser.");
+
+		StringBuilder message = new StringBuilder();
+		if(!routes.isEmpty()) {
+			IRoute route = routes.get(0);
+			IConnection connection = ConnectionsRegistryUtil.safeGetConnectionFor(route);
+			if(connection != null) {
+				message.append("Server: ")
+				.append(connection.getUsername()).append(" ").append(connection.getHost())
+				.append(StringUtils.getLineSeparator());
+			}
+			message.append("Project: ")
+				.append(route.getNamespace())
+				.append(StringUtils.getLineSeparator());
+			//Add more space between server/project info and instruction.
+			message.append(StringUtils.getLineSeparator());
+		}
+		message.append("Select the route to open in a browser.");
+		setMessage(message.toString());
```

Comment:
this looks like a great candidate that can be extracted to its own method, something like #getMessage(IRoute route).
I'll take care of this when merging.